File: 4_Tópicos_avançados/4.02_OpenCV/Ex04/Criar_Imagem.cpp
// Baseado em https://docs.opencv.org/master/db/df5/tutorial_linux_gcc_cmake.html

#include <stdio.h>
#include <opencv2/opencv.hpp>
using namespace std;
using namespace cv;

int main(int argc, char** argv)
{
	int i;
	Mat A(256,256, CV_8U, Scalar(0));
	uchar* p = A.ptr();
	for(i = 0; i < (A.rows*A.cols); ++i)
		p[i] = i;
	imshow("Matriz 256x256", A);
	waitKey(0);
	destroyWindow("Matriz 256x256");

	A = Mat(256, 256, CV_8UC3, Scalar(0));
	p = A.ptr();
	for(i = 0; i < (A.rows*A.cols); ++i)
		p[i*3] = i;
	imshow("Matriz B 256x256 de 3 canais", A);
	waitKey(0);
	destroyWindow("Matriz B 256x256 de 3 canais");

	A = Mat(256, 256, CV_8UC3, Scalar(0));
	p = A.ptr();
	for(i = 0; i < (A.rows*A.cols); ++i)
		p[i*3+1] = i;
	imshow("Matriz G 256x256 de 3 canais", A);
	waitKey(0);
	destroyWindow("Matriz G 256x256 de 3 canais");

	A = Mat(256, 256, CV_8UC3, Scalar(0));
	p = A.ptr();
	for(i = 0; i < (A.rows*A.cols); ++i)
		p[i*3+2] = i;
	imshow("Matriz R 256x256 de 3 canais", A);
	waitKey(0);
	return 0;
}

File: 4_Dicas/4.1_OpenCV/Ex04/Criar_Imagem.py
import numpy, cv2

A = numpy.zeros((256,256), dtype=numpy.uint8)
for k in range(256):
	A[k] = numpy.arange(256)
cv2.imshow("Matriz 256x256", A)
cv2.waitKey(0);
cv2.destroyWindow("Matriz 256x256");

A = numpy.zeros((256,256,3), dtype=numpy.uint8)
for k in range(256):
	A[k,:,0] = numpy.arange(256)
cv2.imshow("Matriz B 256x256 de 3 canais", A)
cv2.waitKey(0);
cv2.destroyWindow("Matriz B 256x256 de 3 canais");

A = numpy.zeros((256,256,3), dtype=numpy.uint8)
for k in range(256):
	A[k,:,1] = numpy.arange(256)
cv2.imshow("Matriz G 256x256 de 3 canais", A)
cv2.waitKey(0);
cv2.destroyWindow("Matriz G 256x256 de 3 canais");

A = numpy.zeros((256,256,3), dtype=numpy.uint8)
for k in range(256):
	A[k,:,2] = numpy.arange(256)
cv2.imshow("Matriz R 256x256 de 3 canais", A)
cv2.waitKey(0);
cv2.destroyWindow("Matriz R 256x256 de 3 canais");